modules/siddhi-core/src/main/java/io/siddhi/core/util/extension/validator/InputParameterValidator.java

```diff
@@ -0,0 +1,134 @@
+     */
+    public static void validateExpressionExecutors(Object objectHavingAnnotation,
+                                                   ExpressionExecutor[] attributeExpressionExecutors)
+            throws SiddhiAppValidationException {
+
+        Extension annotation = objectHavingAnnotation.getClass().getAnnotation(Extension.class);
+        if (annotation == null) {
+            return;
+        }
+        ParameterOverload[] parameterOverloads = annotation.parameterOverloads();
+        Parameter[] parameters = annotation.parameters();
+        String key = AnnotationHelper.createAnnotationKey(annotation);
+
+        //Count the mandatory number of parameters specified in @Extension
+        int mandatoryCount = 0;
+        Map<String, Parameter> parameterMap = new HashMap<>();
+
+        for (Parameter parameter : parameters) {
+            if (!parameter.optional()) {
+                mandatoryCount++;
+            }
+            parameterMap.put(parameter.name(), parameter);
+        }
+
+        //Find the parameterOverLoad
+        ParameterOverload parameterOverload = null;
+        for (ParameterOverload aParameterOverload : parameterOverloads) {
+            String[] overloadParameterNames = aParameterOverload.parameterNames();
+            if (overloadParameterNames.length == attributeExpressionExecutors.length) {
+                boolean isExpectedParameterOverload = true;
+                for (int i = 0; i < overloadParameterNames.length; i++) {
+                    String overloadParameterName = overloadParameterNames[i];
+                    Parameter parameter = parameterMap.get(overloadParameterName);
+                    boolean supportedReturnType = false;
+                    for (DataType type : parameter.type()) {
+                        if (attributeExpressionExecutors[i].getReturnType().toString().
+                                equalsIgnoreCase(type.toString())) {
+                            supportedReturnType = true;
+                            break;
+                        }
+                    }
+                    if (!supportedReturnType) {
+                        isExpectedParameterOverload = false;
+                        break;
+                    }
+                }
+                if (isExpectedParameterOverload) {
+                    parameterOverload = aParameterOverload;
+                    break;
+                }
+            }
+        }
+
+        if (parameterOverload == null) {
+            if (parameterOverloads.length > 0) {
+                List<Attribute.Type> returnTypes = new ArrayList<>();
+                for (ExpressionExecutor expressionExecutor : attributeExpressionExecutors) {
+                    returnTypes.add(expressionExecutor.getReturnType());
+                }
+                throw new SiddhiAppValidationException("There is no parameterOverload for '" + key +
+                        "' that matches attribute types '" + returnTypes + "'.");
```

Comment:
Do we need to throw this exception?

Reply:
When parameterOverloads are given in the extension, the user should only use one of the given overloads. 
In this case, we can't find a matching overload and hence throwing an exception.